sys/cddl/dev/dtrace/dtrace_ioctl.c

```diff
@@ -21,13 +21,19 @@
  * $FreeBSD$
  *
  */
+#include <sys/sysent.h>
+#include <cddl/contrib/opensolaris/uts/common/sys/dtrace.h>
+#include <sys/systm.h>
 
 static int dtrace_verbose_ioctl;
 SYSCTL_INT(_debug_dtrace, OID_AUTO, verbose_ioctl, CTLFLAG_RW,
     &dtrace_verbose_ioctl, 0, "log DTrace ioctls");
 
 #define DTRACE_IOCTL_PRINTF(fmt, ...)	if (dtrace_verbose_ioctl) printf(fmt, ## __VA_ARGS__ )
 
+#define CORRECT_CAP_VERSION(BASE, VAR) \
+	(SV_CURPROC_FLAG(SV_CHERI) ? &BASE.BASE.VAR : &BASE.BASE##_64.VAR)
```

Comment:
Perhaps make the define easier to tie to the fact that it is operating on `*desc_union` types, as in `GET_COMPAT_DESC_FIELD(desc_union, field)` or something like that.